docs/guides/frontend-redux.md

```diff
@@ -0,0 +1,63 @@
+# Frontend Data Architecture
+
+This post describes the architecture used throughout [frontend-scaffold](https://github.com/Code-4-Community/frontend-scaffold), Code4Community's shared frontend template used by all client projects. The scaffold is a React application created with Create React App using the TypeScript template. It uses Redux for application state management. 
+
+## What is Redux? 
+
+Redux is a predictable state container for JavaScript apps.
+
+It helps you write applications that behave consistently, run in different environments (client, server, and native), and are easy to test. On top of that, it provides a great developer experience, such as live code editing combined with a time traveling debugger.
+
+Redux accomplishes this by establishing a set of core principles for accessing, fetching, and updating data in your application. A Redux store has one common *state* object, which you can think of as your "model", to use object-oriented terminology. This model has no "setters", to prevent different parts of the code from updating the data in different ways. 
+
+Instead, the only way to make updates to the store is by dispatching an *action*. Actions contain a type and a payload. The type is a constant identifier, and the payload can be any data type - string, number, boolean, JSON, etc.
+
+Actions and state are linked via functions called *reducers*. A reducer takes a state object and an action object, and returns an updated state object depending on the action. Typically, the reducer will perform a switch statement on different action types. 
+
+For more information, please see [https://redux.js.org/introduction/core-concepts/](https://redux.js.org/introduction/core-concepts/)
```

Comment:
Can you add some images or graphics to try and help explain? I think they could be really useful!